### PR TITLE
Implements a database wrapper for the MySQL database to be used in all database interactions

### PR DESCRIPTION
**Pull request by mfekadu**
# How do I even use this new database wrapper?
see `demo.py`

# What was changed?
* `database_wrapper.py` will replace `db_wrapper.py`
* `config.json` will replace `db_config.py`

# Why it was changed?
Previously our database wrapper was a hodgepodge of functions that ran individual SQL statements. Going forward, the `database_wrapper.py` file contains a wrapper class that will be used for all interactions with the Nimbus database because this will allow us to easily create useful API functions (e.g. `getProfessorInfo`)

# How was it changed?
* `config.json` can be our general configuration file for all things (i.e. DB credentials, API tokens, etc) 
  - [ ] @snekiam can you comment on if the Google Drive API token can go here?

## Technical Tidbits
* `NimbusDatabase` is an abstract class and `NimbusMySQL` inherits `NimbusDatabase`
  * imagine in the future we may switch to use another subclass (e.g. `NimbusPostgres` or `NimbusGoogleDrive`) but so long as that subclass implements the required `@abstractmethod`s then our codebase will remain functional. 

* I followed the [**Template Method**](https://refactoring.guru/design-patterns/template-method) design pattern.

* It was quite [fun to learn about python's `abc` (abstract base class) module](https://stackoverflow.com/questions/3570796/why-use-abstract-base-classes-in-python)

### CODE REVIEW

**Comment by mfekadu** [comment]
resolves #7

**Comment by mfekadu** [comment]
@snekiam is it reasonable to place the Google Drive API **tokens/IDs/keys**  into the `config_SAMPLE.json` file? ... at least as a template?

#10 is the PR that brought up the API key issue

[click here to **review** the current `config_SAMPLE.json` file](https://github.com/calpoly-csai/api/pull/11/files#diff-4636757b28f7050dc3b034749aae63a6)

## e.g. `config_SAMPLE.json`
```json
{
   "google_drive": {
      "ID": "ID",
      "key": "KEY",
      ...
   }
   "mysql": {
     "hostname": "HOSTNAME"
     ...
   }
}
```

**Comment by snekiam** [comment]
> 
> 
> @snekiam is it reasonable to place the Google Drive API **tokens/IDs/keys** into the `config_SAMPLE.json` file? ... at least as a template?
> 
> #10 is the PR that brought up the API key issue
> 
> [click here to **review** the current `config_SAMPLE.json` file](https://github.com/calpoly-csai/api/pull/11/files#diff-4636757b28f7050dc3b034749aae63a6)
> ## e.g. `config_SAMPLE.json`
> 
> ```json
> {
>    "google_drive": {
>       "ID": "ID",
>       "key": "KEY",
>       ...
>    }
>    "mysql": {
>      "hostname": "HOSTNAME"
>      ...
>    }
> }
> ```

Currently, there are two files which have sensitive data in them - one has the folder id, which I think could be easily moved into config_SAMPLE.json.  The other is a settings.yaml file which controls the authentication settings for the google drive library.  I *think* this needs to be done specifically with a yaml file called settings.yaml, but I can see if we could apply keys from the config_SAMPLE.json file.  

There is also a credentials.json file which is generated by the library, and contains access tokens and secrets for accessing google drive as a specific user - this is generated after google account credentials are provided on the first launch from the command line, and definitely needs to be a separate file.

File: database_wrapper.py [hunk 1]
```diff
@@ -5,43 +5,221 @@
...
+    """
+
+    def __init__(self, message: str):
+        self.message = message
+
+
+        
+        
+"""
+utilities.py
```
Comment (snekiam) [inline database_wrapper.py:43]: Should these functions be in a separate file?

Comment (snekiam) [reply database_wrapper.py:43]: that is a utilities.py file

**Review by snekiam — APPROVED** [review]
Looks good, just one comment

**Review by Waidhoferj — APPROVED** [review]
Looks good to me!

File: database_wrapper.py (hunk 1 above)
Comment (mfekadu) [reply database_wrapper.py:43]: yes! TODO! 😅

**Pull request merged**